_posts/2019-10-01-intro-to-hmc/hmc-intro.Rmd

```diff
@@ -0,0 +1,283 @@
+
+This post is directed towards the second group of readers, aiming to make up for the imposition.
+If after reading, you still feel confused, it isn't your fault and _maybe_, not even the fault of this text. Based on an inofficial, minimal-sample-size survey, this topic is a _lot_ more confusing than deep learning :-). As to the content, we'll try to keep it as digestable as possible, while at the same time striving to mention all of the famous incantations: Hamiltonian Monte Carlo, _leapfrog_ steps, NUTS.
+
+
+## So what is it for?
+
+Sampling, or _Monte Carlo_, techniques in general are used when we want to produce samples from, or statistically describe, a distribution we don't have a closed-form formulation of. Sometimes, we might really be interested in the samples; sometimes we just want them so we can compute, for example, the mean and variance of the distribution.
+
+What distribution? In the type of applications we're talking about, we have a _model_, a joint distribution, supposed to describe some reality. Starting from the most basic scenario, it might look like this:
+
+$$
+x \sim \mathcal{Poisson}(\lambda)
+$$
+
+This "joint distribution" only has a single member, a Poisson distribution, that is supposed to model, say, the number of comments in a code review. We also have data on actual code reviews, like this, say ^[data are purely made up]:
+
+```{r, echo = FALSE}
+ggplot(data.frame(x = x), aes(x = x)) + geom_bar() + ggtitle("Number of comments in (fictional) code reviews") + labs(x = "number of comments", y = "number of code reviews") + scale_x_discrete(drop=FALSE)
+```
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body", fig.cap = ""}
+knitr::include_graphics("images/comments.png")
+```
+
+We now want to determine the _parameter_, $\lambda$, of the Poisson that make these data most _likely_. So far, we're not even beign Bayesian yet: There is no prior on this parameter. But of course, we want to be Bayesian, so we add one -- imagine fixed priors ^[elided] on _its_ parameters:
+
+$$
+x \sim \mathcal{Poisson}(\lambda)\\
+\lambda \sim \gamma(\alpha, \beta)\\
+\alpha \sim [...]\\  
+\beta \sim [...]
+$$
+
+This now really being a joint distribution, we have three parameters to determine: $\lambda$, $\alpha$ and $\beta$.
```

Comment:
```suggestion
This being a joint distribution, we have three parameters to determine: $\lambda$, $\alpha$ and $\beta$.
```

Reply:
If there's a concrete way to hint at what lambda, alpha, and beta represent, you might consider adding that. (I don't have a suggestion, because I'm learning this as I read it 😉).

Reply:
I'll do that. The thing is, I deliberately didn't want to go into too much detail here... but let me check if I can find a compromise. See also my general answer upcoming...